File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/csharp/src/de/sots/cellarsandcentaurs/domain/model/HitPoints.cs
namespace de.sots.cellarsandcentaurs.domain.model
{
    public class HitPoints
    {
        private int current;
        private int max;
        private int temporary;

        public HitPoints(int current, int max)
        {
            this.current = current;
            this.max = max;
            this.temporary = 0;
        }

        public static HitPoints init(int max)
        {
            return new HitPoints(max, max, 0);
        }
    }
}

File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/java/src/main/java/de/sots/cellarsandcentaurs/domain/model/HitPoints.java
package de.sots.cellarsandcentaurs.domain.model;

public record HitPoints (
    int current,
    int max,
    int temporary
) {
    public HitPoints(int current, int max) {
        this(current, max, 0);
    }

    public static HitPoints init(int max) {
        return new HitPoints(max, max, 0);
    }
}
